src/OauthTokenFileStorage.php

```diff
@@ -144,8 +142,12 @@ public function removeToken(): void {
   public static function oauthTokenPath() {
     static $token_path;
 
-    $token_path = $token_path
-      ?? rtrim(Settings::get(static::OAUTH_TOKEN_FILE_PATH_SETTINGS_KEY, 'private://.apigee_edge'), " \t\n\r\0\x0B\\/") . '/oauth.dat';
+    if (!isset($token_path)) {
+      // Get the file location from config.
+      $location = \Drupal::config(AuthenticationForm::CONFIG_NAME)->get('oauth_token_storage_location');
+
+      $token_path = (!empty($location) ? rtrim($location, " \t\n\r\0\x0B\\/") : 'private://.apigee_edge') . '/oauth.dat';
```

Comment:
Are you sure rtrim() is necessary here? I do not think the path needs to be sanitized this much.